src/twisted/python/failure.py

```diff
@@ -165,10 +171,17 @@ class Failure(BaseException):
     pickled = 0
     stack = None
 
-    # The opcode of "yield" in Python bytecode. We need this in _findFailure in
-    # order to identify whether an exception was thrown by a
-    # throwExceptionIntoGenerator.
-    _yieldOpcode = chr(opcode.opmap["YIELD_VALUE"])
+    # The opcode of "yield" in Python bytecode. We need this in
+    # _findFailure in order to identify whether an exception was
+    # thrown by a throwExceptionIntoGenerator.
+    # on PY3, b'a'[0] == 97 while in py2 b'a'[0] == b'a' opcodes
+    # are stored in bytes # so we need to properly account for this
+    # difference.
+    if _PY3:
+        _yieldOpcode = opcode.opmap["YIELD_VALUE"]
+    else:
+        _yieldOpcode = chr(opcode.opmap["YIELD_VALUE"])
```

Comment:
I've tested this on Python 2 and Python 3.  This is the right fix, because further down,
we dereference a single element in a bytes array